samples/account-linking/csharp/.tours/token-storage.tour

```diff
@@ -0,0 +1,177 @@
+      "line": 6
+    },
+    {
+      "file": "Source/Services/UserTokenStore/EncryptingUserTokenStoreDecorator.cs",
+      "description": "# Salted protection\r\nThis code protects each token with a symmetric key that is salted with the tenant and user ids. This ensures that if an attacker manages to crack _one_ of the payloads they cannot gain access to all of them. \r\n\r\n",
+      "line": 60,
+      "selection": {
+        "start": {
+          "line": 58,
+          "character": 1
+        },
+        "end": {
+          "line": 60,
+          "character": 64
+        }
+      }
+    },
+    {
+      "file": "Source/Services/UserTokenStore/EncryptingUserTokenStoreDecorator.cs",
+      "description": "This encryption is put together as a decorator around another token storage so that the persistance implementations can focus on the storage and retrieval and re-use this encryption layer. \r\n\r\nIn your own implementations this can be bundled with the persistance layer. ",
+      "line": 62,
+      "selection": {
+        "start": {
+          "line": 62,
+          "character": 9
+        },
+        "end": {
+          "line": 62,
+          "character": 99
+        }
+      }
+    },
+    {
+      "file": "Source/Services/OAuth/OAuthTokenProvider.cs",
+      "description": "# Retriving the Access Token\r\n\r\nTo retrive the stored access token we need to first retrieve the access and refresh tokens",
+      "line": 37
+    },
+    {
+      "file": "Source/Services/OAuth/OAuthTokenProvider.cs",
+      "description": "If the access token that was persisted is still valid, we should re-use it and not use the refresh token",
+      "line": 57
+    },
+    {
+      "file": "Source/Services/OAuth/OAuthTokenProvider.cs",
+      "description": "We attempt to refresh the access token using the refresh token as specified [in the rfc](https://datatracker.ietf.org/doc/html/rfc6749#section-1.5) (step G/H)",
+      "line": 66
+    },
+    {
+      "file": "Source/Services/OAuth/OAuthTokenProvider.cs",
+      "description": "If the refresh _fails_ we need to get the user to re-consent to the application. This is the same as if they haven't consented the first time so we generate a new `state` object and run the consent flow again.",
+      "line": 70,
+      "selection": {
+        "start": {
+          "line": 67,
+          "character": 9
+        },
+        "end": {
+          "line": 70,
+          "character": 10
+        }
+      }
+    },
+    {
+      "file": "Source/Services/OAuth/OAuthTokenProvider.cs",
+      "description": "Before returning, we first persist the new token(s) ",
+      "line": 88,
```

Comment:
Double check these lines and selections, there is some overlap of the content and code, but I think it might have shifted